### PR TITLE
Rework Readersettings screen

### PR DESCRIPTION
**Pull request by CD-Z**
- added sourceID to WebViewReader in body as id for source specific css configuration
- added document picker for CSS and JS files
- added Modal for Custom CSS and Custom JS
- added sourceID to browse screen for users to be able to create custom css

</img></img></img>

### CODE REVIEW

File: src/components/PlusMinusField/PlusMinusField.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+interface PlusMinusFieldProps {
+  labelStyle?: TextStyle | TextStyle[];
+  label: string;
+  value: number;
+  displayValue?: number | string;
+  min?: number;
+  max?: number;
+  valueChange?: number;
+  method: (value: number) => void;
+  onFocus?: (event: Event) => void;
```
Comment (rajarsheechatterjee) [inline src/components/PlusMinusField/PlusMinusField.tsx:15]: Is this being used anywhere?

File: src/components/PlusMinusField/PlusMinusField.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import { IconButtonV2 } from '@components/index';
+interface PlusMinusFieldProps {
+  labelStyle?: TextStyle | TextStyle[];
+  label: string;
+  value: number;
+  displayValue?: number | string;
+  min?: number;
+  max?: number;
+  valueChange?: number;
+  method: (value: number) => void;
```
Comment (rajarsheechatterjee) [inline src/components/PlusMinusField/PlusMinusField.tsx:14]:
```suggestion
  onChange: (value: number) => void;
```

File: src/components/PlusMinusField/PlusMinusField.tsx
Comment (rajarsheechatterjee) [inline src/components/PlusMinusField/PlusMinusField.tsx:1]: You can reference [this component](https://github.com/LNReader/lnreader-rewrite/blob/main/src/components/Counter/Counter.tsx) I had created sometime back.

File: src/screens/browse/components/SourceCard/SourceCard.tsx [hunk 3]
```diff
@@ -33,7 +33,7 @@ const SourceCard: React.FC<Props> = ({
       <View style={styles.details}>
         <Text style={{ color: theme.onSurface }}>{source.sourceName}</Text>
         <Text style={[{ color: theme.onSurfaceVariant }, styles.lang]}>
-          {source.lang}
+          {source.lang + ' ' + source.sourceId}
```
Comment (rajarsheechatterjee) [inline src/screens/browse/components/SourceCard/SourceCard.tsx:36]:
```suggestion
          {`${source.lang} (ID: ${source.sourceId})`}
```

File: src/screens/reader/components/WebViewReader.tsx [hunk 4]
```diff
@@ -214,8 +214,7 @@ const WebViewReader: React.FC<WebViewReaderProps> = props => {
                       }
                     </style>
                   </head>
-                  <body>
-
+                  <body id='source${chapterInfo.sourceId}'>
```
Comment (rajarsheechatterjee) [inline src/screens/reader/components/WebViewReader.tsx:217]:
```suggestion
                  <body id='sourceId-${chapterInfo.sourceId}'>
```

File: src/screens/settings/SettingsReaderScreen/Modals/CustomFileModal.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        copyToCacheDirectory: false,
+        type: mimeType,
+      });
+      if (rawCSS.type === 'success') {
+        let css = await StorageAccessFramework.readAsStringAsync(rawCSS.uri);
+        setCustomFile(css);
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
```
Comment (rajarsheechatterjee) [inline src/screens/settings/SettingsReaderScreen/Modals/CustomFileModal.tsx:57]: You should use `showToast` instead.

File: src/components/PlusMinusField/PlusMinusField.tsx (hunk 1 above)
Comment (CD-Z) [reply src/components/PlusMinusField/PlusMinusField.tsx:15]: Originally I tried something that didn't work and I forgot about this.

**Comment by CD-Z** [comment]
@rajarsheechatterjee I f upped the rebase so I coppied this PR into a new one and close this one.

**Pull request closed without merging**